service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java

```diff
@@ -1304,6 +1375,40 @@ private PaymentRequestDto formPayment(Long orderId, int sumToPay) {
         return paymentRequestDto;
     }
 
+    private int formCertificatesToBeSavedAndCalculateOrderSumClient(OrderFondyClientDto dto,
+        Set<Certificate> orderCertificates,
+        Order order, int sumToPay) {
+        if (sumToPay != 0 && dto.getCertificates() != null) {
+            boolean tooManyCertificates = false;
+            for (String temp : dto.getCertificates()) {
+                if (tooManyCertificates) {
+                    throw new TooManyCertificatesEntered(TOO_MANY_CERTIFICATES);
+                }
+                Certificate certificate = certificateRepository.findById(temp).orElseThrow(
```

Comment:
you're fetching DB for each certificate id . This is not efficient. Create another repository method that accepts set ids instead.